### ISSUE TITLE
CurlHandler empty header values not being sent in the event of a redirect occuring

### ISSUE THREAD

**Issue by martinbooth** [dotnet/corefx#4663]
Thanks for your quick turnaround on https://github.com/dotnet/corefx/issues/4597.

There is still an issue which seems related (although, I'm not personally affected by this one):

The example I gave (http://www.google.com) actually causes a 302 redirect in Australia, and I happened to notice that the header is missing from the subsequent request on linux. This is different behaviour to windows

Linux:
```
GET http://www.google.com/ HTTP/1.1
Host: www.google.com
Accept: */*
Accept-Encoding: gzip,deflate
Connection: Keep-Alive
No-value: 
With-value: value
```
```
GET http://www.google.com.au/?gfe_rd=cr&ei=btBTVp6vCsTu8we45Kv4Dg HTTP/1.1
Host: www.google.com.au
Accept: */*
Accept-Encoding: gzip,deflate
Connection: Keep-Alive
With-value: value
```

Windows:

```
GET http://www.google.com/ HTTP/1.1
Accept-Encoding: gzip, deflate
No-value: 
With-value: value
Connection: Keep-Alive
Host: www.google.com
```
```
GET http://www.google.com.au/?gfe_rd=cr&ei=pdBTVqyuG8Tu8we45Kv4Dg HTTP/1.1
Accept-Encoding: gzip, deflate
No-value: 
With-value: value
Connection: Keep-Alive
Host: www.google.com.au
```

**Comment by stephentoub** [comment]
Thanks, @martinbooth.  We'll take a look.

**Comment by vijaykota** [comment]
Fixed in #4683
